app/controllers/admin_controller.rb

```diff
@@ -1,5 +1,17 @@
 # Displays the admins' homepage
 class AdminController < ApplicationController
+  before_action :authorize
+
   def index
   end
+
+
+  protected
+
+  def authorize
+    unless User.find_by(id: session[:user_id])
+      redirect_to login_url, notice: 'Please log in to access this page'
+    end
+  end
```

Comment:
Suggestion: the ApplicationController could have a before_action that looks up the user and puts it in a @user variable. Then this authorize method could just check whether the @user exists.